tests/includes/wait-for.sh

```diff
@@ -1,31 +1,34 @@
-# SHORT_TIMEOUT creates a consistent short timeout of the wait_for condition.
+#!/bin/bash
+# SHORT_TIMEOUT is the time between polling attempts.
 SHORT_TIMEOUT=5
 
 # wait_for defines the ability to wait for a given condition to happen in a
 # juju status output. The output is JSON, so everything that the API server
 # knows about should be valid.
 # The query argument is a jq query.
-# The default timeout is 10 minutes (120 attempts @ 5 seconds each).
-# You can change this by providing the max_attempts argument.
+# The default timeout is 10 minutes. You can change this by providing the
+# timeout argument (an integer number of seconds).
 #
 # ```
-# wait_for <model name> <query> [<max_attempts>]
+# wait_for <model name> <query> [<timeout>]
 # ```
 wait_for() {
-	local name query max_attempts
+	local name query timeout
 
 	name=${1}
 	query=${2}
-	max_attempts=${3:-120} # default timeout: 120x5s = 10m
+	timeout=${3:-600} # default timeout: 600s = 10m
 
 	attempt=0
+	start_time="$(date -u +%s)"
 	# shellcheck disable=SC2046,SC2143
 	until [[ "$(juju status --format=json 2>/dev/null | jq -S "${query}" | grep "${name}")" ]]; do
 		echo "[+] (attempt ${attempt}) polling status for" "${query} => ${name}"
 		juju status --relations 2>&1 | sed 's/^/    | /g'
 		sleep "${SHORT_TIMEOUT}"
 
-		if [[ "${attempt}" -ge ${max_attempts} ]]; then
+    elapsed=$(date -u +%s)-$start_time
```

Comment:
I don't think this is doing what you intend. [Bash is weird](https://unix.stackexchange.com/questions/93029/how-can-i-add-subtract-etc-two-numbers-with-bash). I think it will set `elapsed` to the literal string `newtime-oldtime`:

```
$ start_time="$(date -u +%s)"
$ elapsed=$(date -u +%s)-$start_time
$ echo $elapsed 
1658463345-1658463334
```

I guess these functions don't have tests? In which case we'll need to manually test a couple of different cases.

Or maybe I'm missing something!

Reply:
My bad, you're right, sorry! Bash is still weird. :-)

```
0 ~/w/juju/tests$ echo $elapsed
1658466194-1658466184
0 ~/w/juju/tests$ timeout=9
0 ~/w/juju/tests$ [[ "${elapsed}" -ge ${timeout} ]]
0 ~/w/juju/tests$ timeout=10
0 ~/w/juju/tests$ [[ "${elapsed}" -ge ${timeout} ]]
0 ~/w/juju/tests$ timeout=11
0 ~/w/juju/tests$ [[ "${elapsed}" -ge ${timeout} ]]
1 ~/w/juju/tests$
```

(Weird that `timeout=10` returns 0 as it's "equal to" the difference so I would think it'd return 1.)

Reply:
Yeah `-ge` is greater than or equal. Remember that 0 means success and 1 means failure (i.e. the opposite of booleans)